plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployCommandHandler.java

```diff
@@ -41,22 +51,25 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
     try {
       IProject project = helper.getProject(event);
       if (project != null) {
-        launchDeployJob(project);
+        launchDeployJob(project, new SameShellProvider(HandlerUtil.getActiveShell(event)));
       }
       // return value must be null, reserved for future use
       return null;
-    } catch (CoreException coreException) {
-      throw new ExecutionException(Messages.getString("deploy.failed.error.message"), coreException); //$NON-NLS-1$
+    } catch (CoreException | IOException exception) {
+      throw new ExecutionException(Messages.getString("deploy.failed.error.message"), exception); //$NON-NLS-1$
     }
   }
 
-  private void launchDeployJob(IProject project) {
-    String now = Long.toString(System.currentTimeMillis());
+  private void launchDeployJob(IProject project, IShellProvider shellProvider) throws IOException, CoreException {
+    IPath workDirectory = createWorkDirectory();
+    
+    logInAndSaveCredential(workDirectory, shellProvider);
```

Comment:
So the only thing I'd change is to explicitly pass along the credentials filename to the deploy job rather than having it pick up the credential from some fixed-name file.  

Even better would move responsibility for the writing the credentials to disk to DeployJob — it has more control over the lifecycle of the credentials file (i.e., can delete it as soon as possible).  And gcloud could hypothetically take the credentials as an envvar.  (My earlier issue was with the possibility of deploy triggering UI for login.)

Having DeployJob write out to disk also avoids the possibility of multiple deploy jobs writing out conflicting credentials.